src/affix.vue

```diff
@@ -169,6 +169,12 @@ export default {
         >= this.relativeElement.offsetHeight;
 
       if (affixIsBiggerThanRelativeElement) {
```

Comment:
Actually, I think my change introduces another issue: when the element is hidden, `this.relativeElement.offsetHeight === 0` and in that case we should not change the current state. What do you think about adding a check on `this.relativeElement.offsetParent`?

```javascript
// ...
if (affixIsBiggerThanRelativeElement) {
  if (this.relativeElement.offParent) {
    if (this.scrollAffix && this.currentScrollAffix !== 'scrollaffix-top') {
      this.setScrollAffixTop();
    } else if (this.currentState !== 'affix-top') {
      this.setAffixTop();
    }
  }

  return;
}
// ...
```

or maybe that check should be on the overall affix logic:

```javascript
onScroll() {
   //...
  this.updateData();

  if (!this.relativeElement.offParent) {
     // relative element not displayed
     return;
  }

  const affixIsBiggerThanRelativeElement = ...
  // ...
}
```